src/loader/book.rs

```diff
@@ -0,0 +1,322 @@
+#![allow(missing_docs, unused_variables, unused_imports, dead_code)]
+
+use std::path::Path;
+use std::collections::VecDeque;
+use std::fs::File;
+use std::io::Read;
+
+use loader::summary::{Summary, Link, SummaryItem, SectionNumber};
+use errors::*;
+
+
+/// A dumb tree structure representing a book.
+///
+/// For the moment a book is just a collection of `BookItems`.
+#[derive(Debug, Clone, Default, PartialEq, Serialize, Deserialize)]
+pub struct Book {
+    /// The sections in this book.
+    pub sections: Vec<BookItem>,
+}
+
+impl Book {
+    /// Create an empty book.
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Get a depth-first iterator over the items in the book.
+    pub fn iter(&self) -> BookItems {
+        BookItems { items: self.sections.iter().collect() }
+    }
+}
+
+/// Enum representing any type of item which can be added to a book.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum BookItem {
```

Comment:
I'm not sure why we need separation between `Chapter`, `SummaryItem` and `BookItem`